app/models.py

```diff
@@ -20,14 +22,112 @@
 from .env import load_env
 from .json_util import to_json
 
-session_context_id = contextvars.ContextVar("request_id", default=None)
-DBSession = scoped_session(sessionmaker(), scopefunc=session_context_id.get)
-
 env, cfg = load_env()
 strict = cfg["security.strict"]
 use_webhook = cfg["security.slack.enabled"]
 webhook_url = cfg["security.slack.url"]
 
+session_context_id = contextvars.ContextVar("request_id", default=None)
+# left here for backward compatibility:
+DBSession = scoped_session(sessionmaker(), scopefunc=session_context_id.get)
+
+
+@contextmanager
+def Session(user_or_token=None, use_auto_verify=True):
+    """
+    Generate a scoped session that also has knowledge
+    of the current user, and that can automatically
+    verify and commit the changes to it when going out of context.
+
+    Parameters
+    ----------
+    user_or_token: baselayer.app.models.User object
+        or baselayer.app.models.Token object.
+        The object representing the current user.
+        Can be None only if the use_auto_verify
+        parameter is set to False.
+    use_auto_verify: boolean
+        if True (default), will call verify and commit
+        functions of the session before exiting the context.
+        If True, must specify a legal User object.
+
+    Returns
+    -------
+    a scoped session object that can be used in a context
+    manager to access the database. If auto verify is enabled,
+    will use the current user given to apply verification
+    and commit to the database when exiting context.
+
+    """
+
+    def verify(self):
+        # import here to avoid circular import loop
+        from .handlers.base import bulk_verify
```

Comment:
Why not move `bulk_verify` to a more sensible place; it does not need to live on the handler.

Reply:
You may be right. I was just trying not to change more than I had to. I will give it a try.